### PR TITLE
Datastore : Fix StructuredQuery.Builder - Allow limit of 0

### PR DESCRIPTION
**Pull request by athakor**

### CODE REVIEW

File: google-cloud-clients/google-cloud-datastore/src/test/java/com/google/cloud/datastore/DatastoreTest.java [hunk 1]
```diff
@@ -691,6 +691,27 @@ public void testRunKeyQueryWithOffset() {
...
+    datastore.put(ENTITY1, ENTITY2);
+    Query<Key> keyQuery = Query.newKeyQueryBuilder().setLimit(2).build();
+    QueryResults queryResults = datastore.run(keyQuery);
+    int resultCount = 0;
+    while (queryResults.hasNext()) {
+      queryResults.next();
+      resultCount++;
+    }
+    assertEquals(2, resultCount);
+  }
```
Comment (rahulKQL) [inline google-cloud-clients/google-cloud-datastore/src/test/java/com/google/cloud/datastore/DatastoreTest.java:714]: Does below works? If yes then can you please update it

```java
@Test
  public void testRunKeyQueryWithLimit() {
    datastore.put(ENTITY1, ENTITY2);
    Query<Key> keyQuery = Query.newKeyQueryBuilder().setLimit(2).build();
    QueryResults queryResults = datastore.run(keyQuery);
    int resultCount = 0;
    while (queryResults.hasNext()) {
      queryResults.next();
      resultCount++;
    }
    assertEquals(2, resultCount);
   Query<Key> query = Query.newKeyQueryBuilder().setLimit(0).build();
    QueryResults results = datastore.run(query);
    int count = 0;
    while (results.hasNext()) {
      results.next();
      count++;
    }
    assertEquals(0, count);
  }
```

reason being even though there are entities in datastore after setting limit to 0 u r getting 0 as results.

File: google-cloud-clients/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java [hunk 2]
```diff
@@ -858,4 +858,17 @@ public void testSkippedResults() {
...
+    Query query = Query.newEntityQueryBuilder().setLimit(0).build();
+    QueryResults results = DATASTORE.run(query);
+    assertFalse(results.hasNext());
+
+    DATASTORE.put(ENTITY1);
+    Query<Key> keyQuery = Query.newKeyQueryBuilder().setLimit(1).build();
+    QueryResults queryResults = DATASTORE.run(keyQuery);
+    assertTrue(queryResults.hasNext());
+    assertEquals(KEY1, queryResults.next());
+  }
```
Comment (rahulKQL) [inline google-cloud-clients/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java:873]: same here

**Review by rahulKQL — CHANGES_REQUESTED** [review]

File: google-cloud-clients/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java (hunk 2 above)
Comment (athakor) [reply google-cloud-clients/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java:873]: done. working fine even if we change sequence of code

File: google-cloud-clients/google-cloud-datastore/src/test/java/com/google/cloud/datastore/DatastoreTest.java (hunk 1 above)
Comment (athakor) [reply google-cloud-clients/google-cloud-datastore/src/test/java/com/google/cloud/datastore/DatastoreTest.java:714]: done. working fine even if we change sequence of code

**Review by rahulKQL — APPROVED** [review]

**Pull request closed without merging**